pr_agent/tools/pr_similar_issue.py

```diff
@@ -40,8 +36,7 @@ def __init__(self, issue_url: str, args: list = None):
             environment = get_settings().pinecone.environment
         except Exception:
             if not self.cli_mode:
-                repo_name, original_issue_number = self.git_provider._parse_issue_url(self.issue_url.split('=')[-1])
-                issue_main = self.git_provider.repo_obj.get_issue(original_issue_number)
+                issue_main = self.git_provider.parse_issue_url_and_get_issue(self.issue_url.split('=')[-1])
```

Comment:
**Suggestion:** Consider using a more descriptive variable name than `issue_main`. This will improve code readability. [readability]
```suggestion
                main_issue = self.git_provider.parse_issue_url_and_get_issue(self.issue_url.split('=')[-1])
```